great_expectations/metrics/metric.py

```diff
@@ -73,17 +73,17 @@ def __new__(cls, *args, **kwargs):
         cls.name = cls._get_metric_name()
         return super().__new__(cls)
 
-    @root_validator(pre=True)
-    @classmethod
-    def _set_config(cls, values) -> dict:
-        if "config" not in values or values["config"] is None:
-            values["config"] = cls._to_config(values)
-        return values
-
     @property
     def id(self) -> MetricConfigurationID:
         return self.config.id
 
+    @property
```

Comment:
You could cache here also? That makes it clear that is fixed per instance.

Reply:
That's what I tried at first, but you actually shouldn't do it according to [this ruff rule we have](https://docs.astral.sh/ruff/rules/cached-instance-method/). That's why I had to make it a `staticmethod` and pass the class instead of using `self`.